src/lib/pipes/pipe.js

```diff
@@ -1,20 +1,251 @@
+import path from 'path';
 import Utils from "../utils.js";
+import Settings from '../settings.js';
 
 class Pipe {
   constructor(rawData, normalisedEvents) {
-    this.rawData = rawData;
+    // rawMeta is an entire database row from the raw_data table, minus data field
+    // rawData is the contents of the data field
+    const { data, ...meta } = rawData;
+    this.rawMeta = meta;
+    if(data == null){
+      this.rawData = {};
+    }else{
+      this.rawData = data;
+    }
+    this.provenance = {};
```

Comment:
What is this doing if we add to raw_data anyway? (Not totally sure adding to raw_data is best way anyway, maybe have some extra fields you pass to NormalisedEvent class? Can leave in for now, but I would have taken this and provanence function out totally

Reply:
Yeah sorry this was leftover from a previous slightly different way of doing the provenance. Taken it out with recent commit.